clang/lib/CIR/CodeGen/CIRGenExprComplex.cpp

```diff
@@ -108,7 +108,7 @@ class ComplexExprEmitter : public StmtVisitor<ComplexExprEmitter, mlir::Value> {
   mlir::Value VisitObjCMessageExpr(ObjCMessageExpr *E) {
     llvm_unreachable("NYI");
   }
-  mlir::Value VisitArraySubscriptExpr(Expr *E) { llvm_unreachable("NYI"); }
+  mlir::Value VisitArraySubscriptExpr(Expr *E) { return emitLoadOfLValue(E); }
```

Comment:
Can we please keep same format as in upstream?

Looking to classical codegen it is like here (in-class-body methot). So, it might be better to update upstream to align more.